In [28]:
import numpy as np
from scipy.stats import t

In [30]:
data=[1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
n=len(data)

**99% Confidence interval using sample standard deviation by t_distribution**

In [33]:
confidence=0.99
alpha=1-confidence

mean=np.mean(data)
s=np.std(data,ddof=1)
#ddof=0 means divides by n which is population standard deviation and ddof=1 means divided by n-1 which is sample standard deviation denominator

t_critical=t.ppf(1-alpha/2,df=n-1)
margin_error=t_critical*s/np.sqrt(n)

lower_CI=mean-margin_error
upper_CI=mean+margin_error

print(f"Mean:{mean:.3f}")
print(f"Standard deviation:{s:.3f}")

print(f"99% confidence interval using sample standard deviation is :{lower_CI:.3f} to {upper_CI:.3f}")

Mean:1.239
Standard deviation:0.193
99% confidence interval using sample standard deviation is :1.090 to 1.387


**Why we are using t_distribution**

- Population standard deviation is unknown
- Sample size is small(n<30)
  
Therefore we use t-distribution

**Steps to Construct the Confidence Interval**

- Collect and examine the sample data
- Calculate sample statistics(Mean and standard deviation)
- Select appropiate distribution
- Calculate the critical value
- Compute margin of error
- Calculate the confidence interval
- Interpret the result
We are 99% confident that the true mean durability of the print-heads lies between 1.090 and 1.387 million characters.

**Rationale for Using the t-Distribution**

- Unknown population standard deviation: When σ is not known, using standard deviation introduces extra variability.
- Small sample size: The smaller the sample, the more uncertain our estimate of the population mean.
- t-distribution properties:
    - It is similar to the normal distribution but wider (more spread in the tails).
    - The width of the t-distribution decreases as sample size increases, eventually approaching the normal distribution when n is large.
    - Using the t-distribution ensures the confidence interval accurately reflects the uncertainty due to a small sample and unknown σ.

**99% confidence interval using population standard deviation by t_distribution**


In [37]:
from scipy.stats import norm

In [39]:
sigma=0.2
alpha=1-confidence

z_critical=norm.ppf(1-alpha/2)
margin_error=z_critical*sigma/np.sqrt(n)

lower_CI=mean-margin_error
upper_CI=mean+margin_error

print(f"99% confidence interval using population standard deviation is :{lower_CI:.3f} to {upper_CI:.3f}")

99% confidence interval using population standard deviation is :1.106 to 1.372
